From: coolsnowwolf <coolsnowwolf@gmail.com>
Subject: [PATCH] nss-ecm: add Linux 6.18 build support

Linux 6.18 no longer consumes EXTRA_CFLAGS in the top-level Kbuild.
Add the package-provided NSS include paths to the module flags.

Signed-off-by: coolsnowwolf <coolsnowwolf@gmail.com>
---
--- a/Makefile
+++ b/Makefile
@@ -6,6 +6,7 @@
 endif
 
 obj-m += ecm.o
+ccflags-y += $(EXTRA_CFLAGS)
 ifeq ($(BUILD_ECM_WIFI_PLUGIN),y)
 obj-m += ecm_wifi_plugins/
 endif
--- a/ecm_db/ecm_db_timer.c
+++ b/ecm_db/ecm_db_timer.c
@@ -521,5 +521,9 @@
 	 * as we would wait for the callback to finish and it would wait
 	 * indefinately for the lock to be released!
 	 */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 18, 0))
+	timer_delete_sync(&ecm_db_timer);
+#else
 	del_timer_sync(&ecm_db_timer);
+#endif
 }
--- a/frontends/nss/ecm_nss_ported_ipv4.c
+++ b/frontends/nss/ecm_nss_ported_ipv4.c
@@ -1210,11 +1210,15 @@
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 13, 0))
 			uint32_t tcp_be_liberal = nf_ct_tcp_be_liberal;
 			uint32_t tcp_no_window_check = nf_ct_tcp_no_window_check;
-#else
+	#elif (LINUX_VERSION_CODE < KERNEL_VERSION(6, 18, 0))
 			struct nf_tcp_net *tn = nf_tcp_pernet(nf_ct_net(ct));
 			uint32_t tcp_be_liberal = tn->tcp_be_liberal;
 			uint32_t tcp_no_window_check = tn->tcp_no_window_check;
-#endif
+	#else
+			struct nf_tcp_net *tn = nf_tcp_pernet(nf_ct_net(ct));
+			uint32_t tcp_be_liberal = tn->tcp_be_liberal;
+			uint32_t tcp_no_window_check = 0;
+	#endif
 			ecm_db_connection_address_get(feci->ci, ECM_DB_OBJ_DIR_FROM, addr);
 			ecm_front_end_flow_and_return_directions_get(ct, addr, 4, &flow_dir, &return_dir);
 
--- a/frontends/nss/ecm_nss_ported_ipv6.c
+++ b/frontends/nss/ecm_nss_ported_ipv6.c
@@ -1130,11 +1130,15 @@
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 13, 0))
 			uint32_t tcp_be_liberal = nf_ct_tcp_be_liberal;
 			uint32_t tcp_no_window_check = nf_ct_tcp_no_window_check;
-#else
+	#elif (LINUX_VERSION_CODE < KERNEL_VERSION(6, 18, 0))
 			struct nf_tcp_net *tn = nf_tcp_pernet(nf_ct_net(ct));
 			uint32_t tcp_be_liberal = tn->tcp_be_liberal;
 			uint32_t tcp_no_window_check = tn->tcp_no_window_check;
-#endif
+	#else
+			struct nf_tcp_net *tn = nf_tcp_pernet(nf_ct_net(ct));
+			uint32_t tcp_be_liberal = tn->tcp_be_liberal;
+			uint32_t tcp_no_window_check = 0;
+	#endif
 			ecm_front_end_flow_and_return_directions_get(ct, src_ip, 6, &flow_dir, &return_dir);
 
 			DEBUG_TRACE("%px: TCP Accel Get window data from ct %px for conn %px\n", feci, ct, feci->ci);
